doc/nrf/gs_modifying.rst

```diff
@@ -14,27 +14,53 @@ After programming and testing an application, you probably want to make some mod
 Adding files and changing compiler settings
 *******************************************
 
-All files that your application uses must be specified in the :file:`CMakeLists.txt` file.
-By default, most samples include only the main application file :file:`src/main.c`.
-This means that you must add all other files that you are using.
+The |NCS| build system is based on Zephyr, whose build system is based on `CMake`_.
+For more information, see :ref:`zephyr:application` in the Zephyr documentation.
 
-You can also configure compiler options, application defines, or include directories, or set :ref:`build types <gs_modifying_build_types>` in :file:`CMakeLists.txt`.
+In the |NCS|, the application is a CMake project.
+As such, the application controls the configuration and build process of both itself, Zephyr, and sourced libraries.
+The application's :file:`CMakeLists.txt` file is the main CMake project file and the source of the build process configuration.
+
+Zephyr creates a CMake component that must be imported by the application into its :file:`CMakeLists.txt` file.
+When imported, the application is able to reference items provided by both Zephyr and the |NCS|.
+
+Importing the Zephyr `CMake`_ component creates the ``app`` library.
+All files that your application uses must be added to this library.
```

Comment:
What do you mean by this ?

Can I not create additional libraries and link those to the `app` lib ?

Reply:
I meant only the source files that are added to the application by the user. These are added to the app library.
Naturally user could create a new library (make it get everything needed from Zephyr) and link it with app library.

We could add a passage about that but if we want to cover adding new application libraries I would add a separate (advanced) section. (taking into account that user much know something about cmake to not create more problems this way)

Reply:
Then I would rephrase it, especially remove the word **must**.
How about:
```suggestion
 All files that your application uses can be added to this library. 
```